homeassistant/components/eight_sleep/sensor.py

```diff
@@ -179,6 +183,9 @@ def __init__(
         elif self._sensor in ("current_sleep", "last_sleep", "current_sleep_fitness"):
             self._attr_native_unit_of_measurement = "Score"
 
+        if self._sensor != "sleep_stage":
+            self._attr_state_class = SensorStateClass.MEASUREMENT
+
         _LOGGER.debug(
```

Comment:
It might make this a bit more manageable to make SensorEntityDescriptions for user sensors in a future PR

Reply:
agreed, I want to split this out so the sensor logic is easier to understand but definitely belongs in a new PR